### PR TITLE
[SPARK-19946][testing]DebugFilesystem.assertNoOpenStreams should report the open streams to help debugging

### PR DESCRIPTION
**Pull request by bogdanrdc**
## What changes were proposed in this pull request?

DebugFilesystem.assertNoOpenStreams throws an exception with a cause exception that actually shows the code line which leaked the stream. 

## How was this patch tested?
New test in SparkContextSuite to check there is a cause exception.

### CODE REVIEW

**Comment by hvanhovell** [comment]
@bogdanrdc can you add the ticket to the title?

File: core/src/test/scala/org/apache/spark/SparkContextSuite.scala [hunk 1]
```diff
@@ -537,6 +539,21 @@ class SparkContextSuite extends SparkFunSuite with LocalSparkContext with Eventu
       }
     }
   }
+  test("SPARK-19446: DebugFilesystem.assertNoOpenStreams should report " +
```
Comment (rxin) [inline core/src/test/scala/org/apache/spark/SparkContextSuite.scala]: add a blank line here ...

File: core/src/test/scala/org/apache/spark/DebugFilesystem.scala [hunk 2]
```diff
@@ -44,7 +44,8 @@ object DebugFilesystem extends Logging {
         logWarning("Leaked filesystem connection created at:")
         exc.printStackTrace()
       }
-      throw new RuntimeException(s"There are $numOpen possibly leaked file streams.")
+      throw new RuntimeException(s"There are $numOpen possibly leaked file streams.",
```
Comment (srowen) [inline core/src/test/scala/org/apache/spark/DebugFilesystem.scala]: This is a style nit, but while we're here, I wouldn't mind avoiding RuntimeException. IllegalStateException is more specific and canonical. No big deal, just if you're making another change anwyay.

**Comment by hvanhovell** [comment]
retest this please

**Comment by hvanhovell** [comment]
LGTM - merging to master. Thanks!

**Pull request closed without merging**